subsys/bluetooth/services/ias.c

```diff
@@ -0,0 +1,122 @@
+/** @file
+ *  @brief IAS Service sample
+ */
+
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "net/buf.h"
+#include "bluetooth/services/ias.h"
+#include <init.h>
+#include <zephyr.h>
+#include <logging/log.h>
+#include <bluetooth/gatt.h>
+
+#define LOG_LEVEL CONFIG_BT_IAS_LOG_LEVEL
+LOG_MODULE_REGISTER(ias);
+
+#define ALERT_MAX_LEN 1
+#define BT_IAS_ALERT_LVL_NO_ALERT		0
+#define BT_IAS_ALERT_LVL_MILD_ALERT		1
+#define BT_IAS_ALERT_LVL_HIGH_ALERT		2
+
+#define GATT_PERM_WRITE_MASK    (BT_GATT_PERM_WRITE | \
+				 BT_GATT_PERM_WRITE_ENCRYPT | \
+				 BT_GATT_PERM_WRITE_AUTHEN)
+
+#ifndef CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN
+#define CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN 0
+#endif
+#ifndef CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT
+#define CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT 0
+#endif
+
+#define IAS_GATT_PERM_DEFAULT (				\
+	CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN ?	\
+	(BT_GATT_PERM_WRITE_AUTHEN) :			\
+	CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT ?	\
+	(BT_GATT_PERM_WRITE_ENCRYPT) :			\
+	(BT_GATT_PERM_WRITE))					\
```

Comment:
Since we already require the `#ifndef` for this to work, would it have been simpler with

```c
#if defined(CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN)
#define IAS_GATT_PERM_DEFAULT BT_GATT_PERM_WRITE_AUTHEN
#elif defined(CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT)
#define IAS_GATT_PERM_DEFAULT CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT
#else
#define IAS_GATT_PERM_DEFAULT BT_GATT_PERM_WRITE
#endif
```
?